vulture_os/toolkit/api_parser/cisco_umbrella/cisco_umbrella.py

```diff
@@ -0,0 +1,173 @@
+        super().__init__(data)
+
+        self.cisco_umbrella_client_id = data["cisco_umbrella_client_id"]
+        self.cisco_umbrella_secret_key = data["cisco_umbrella_secret_key"]
+
+        self.cisco_umbrella_access_token = data.get("cisco_umbrella_access_token", None)
+        self.cisco_umbrella_expires_at = data.get("cisco_umbrella_expires_at", None)
+
+        self.session = None
+
+    def _get_token(self):
+        auth = HTTPBasicAuth(self.cisco_umbrella_client_id, self.cisco_umbrella_secret_key)
+        client = BackendApplicationClient(client_id=self.cisco_umbrella_client_id)
+        oauth = OAuth2Session(client=client)
+        token = oauth.fetch_token(token_url=self.TOKEN_URL, auth=auth)
+        self.cisco_umbrella_access_token = token.get("access_token")
+        self.cisco_umbrella_expires_at = datetime.fromtimestamp(token.get("expires_at")/1000, tz=timezone.now().astimezone().tzinfo)
+
+    def _connect(self):
+        try:
+            # Check for expiration with 10 seconds difference to be sure token will still be valid for some time
+            if not self.cisco_umbrella_access_token or not self.cisco_umbrella_expires_at or (self.cisco_umbrella_expires_at - timedelta(seconds=10)) < timezone.now():
+                self._get_token()
+
+            if self.session is None:
+                self.session = requests.Session()
+
+                headers = self.HEADERS
+                headers.update({'Authorization': f"Bearer {self.cisco_umbrella_access_token}"})
+
+                self.session.headers.update(headers)
+
+        except Exception as err:
+            raise CiscoUmbrellaAPIError(err)
+
+    def __execute_query(self, method, url, query, timeout=30):
+        '''
+        raw request doesn't handle the pagination natively
+        '''
+        self._connect()
+
+        response = self.session.get(url,
+            params=query,
+            headers=self.HEADERS,
+            timeout=timeout,
+            proxies=self.proxies,
+            verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl
+        )
+
+        # response.raise_for_status()
+        if response.status_code != 200:
+            raise CiscoUmbrellaAPIError(f"Error at Cisco-Umbrella API Call URL: {url} Code: {response.status_code} Content: {response.content}")
+
+        return response.json()
+
+    def test(self):
+        try:
+            result = self.get_logs(timezone.now()-timedelta(hours=2), timezone.now())
+
+            return {
+                "status": True,
+                "data": result
+            }
+        except Exception as e:
+            logger.exception(f"[{__parser__}]:test: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def get_logs(self, since, to, index=0):
+
+        payload = {
+            'offset': index,
+            'limit': 5000,
```

Comment:
```suggestion
            'limit': self.LIMIT_MAX,
```
If you create a constant, use it ;)